'Rename' rake task rewrite

xtrasimplicity opened the pull request:
I had some issues with running the 'rename' rake task on OS X sierra (it looks as if 'rename' doesn't use the GNU rename utility), so re-wrote the task to use Ruby rather than Bash. 

Please let me know what you think, and feel free to make changes if/as required. :)

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dyson/skeleton_ruby/1)

<!-- Reviewable:end -->

xtrasimplicity committed:
Rewrote the rename rake task, so that it uses Ruby rather than Bash (as it didn't work too well on OS X).

xtrasimplicity committed:
Improved code such that it is more consistent. Prevented the "rename" task from modifying itself.

xtrasimplicity committed:
Updated Readme file.

Updated the Readme file to reflect the changes made to the Rename task.

dyson commented:
Thanks Andrew! Great to remove the rename dependencies. It always felt a little hackish. I'll have a proper look over it when I get a chance.

Just one question about why you have excluded the rename rake task from modifying itself?

The rename task should be able to be ran multiple times so you can rename the application over and over again which requires the task to modify itself. I wonder if there is a better way to get the module name dynamically and then we don't have to have it hard coded?

xtrasimplicity committed:
Allowed the rename rake task to rename occurrences of ModuleName or module_name in the task itself.

xtrasimplicity committed:
Merge branch 'master' of https://github.com/xtrasimplicity/skeleton_ruby

xtrasimplicity committed:
Removed hard-coded references to SkeletonRuby and skeleton_ruby in tasks/rename.rake, and replaced them with constants.

xtrasimplicity commented:
I can't remember exactly why I decided to prevent the rename task from renaming itself; but I agree, it would be beneficial to be able to run it as many times as you'd like. I've added a commit to reflect this change.

Given the rake task now re-writes all occurrences of this string, I don't think it'd be worth the time/complexity trying to determine the current module name dynamically (given that the rake task doesn't run under the same 'context' as the application). I've switched references to SkeletonRuby and skeleton_ruby to constants, rather than hard-coded strings, which will make it a little easier to work with, and would arguably reduce the execution time of the task (albeit most likely only by a matter of milliseconds).
